client/state/login/reducer.js

```diff
@@ -78,6 +82,20 @@ export const twoFactorAuthPushPoll = createReducer( { inProgress: false, success
 	[ TWO_FACTOR_AUTHENTICATION_PUSH_POLL_COMPLETED ]: state => ( { ...state, inProgress: false, success: true } ),
 } );
 
+export const socialAccount = createReducer( { creating: false }, {
+	[ SOCIAL_CREATE_ACCOUNT_REQUEST ]: () => ( { creating: true } ),
+	[ SOCIAL_CREATE_ACCOUNT_FAILURE ]: ( state, { error } ) => ( { creating: false, createError: error } ),
+	[ SOCIAL_CREATE_ACCOUNT_SUCCESS ]: ( state, { data: { username, bearerToken } } ) => ( {
+		creating: false,
+		username,
+		bearerToken
```

Comment:
It's not great to have the bearerToken in the `socialAccount` state, that's why I didn't move it to redux initially. Maybe we should do 1 action to create the social account and log in. However, it would need a bit of refactoring around the `WpcomLoginForm` component which is not welcomed in this PR.